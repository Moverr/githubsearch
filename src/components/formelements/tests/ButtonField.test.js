import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
 
import Buttonfield from '../ButtonField';
 


 
describe('Testing Input FIeld', () => {
  it('should  render correctly', () => {    
    const props = { };

      const wrapper = shallow(<Buttonfield props />);
      expect(wrapper).toMatchSnapshot();
  });

  it(' should respond to click event ',() =>{
      const mockCallBAck = jest.fn();
      const props = {
        title:"Button Title",
        name:'Button Name',
        message:'Button Message',
        className:'buttonField',
        id:"buttonID",
        callback:mockCallBAck()
    };
      const button = shallow(<Buttonfield props ><i>movers</i></Buttonfield>);
      button.find('button').simulate('click'); 
     expect(mockCallBAck).toHaveBeenCalled();      
     expect(mockCallBAck).toHaveBeenCalledTimes(1);
});

})



 