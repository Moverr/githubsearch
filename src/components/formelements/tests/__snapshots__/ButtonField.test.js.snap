// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button FIeld should  render correctly 1`] = `ShallowWrapper {}`;

exports[`Testing Input FIeld should  render correctly 1`] = `ShallowWrapper {}`;
